### DISCUSSION TITLE
WinUI3: How to handle custom window message?

### DISCUSSION THREAD

**Discussion by gscheartA** [microsoft/microsoft-ui-xaml#7627] [Q&A]
In WPF, I can ComponentDispatcher.ThreadPreprocessMessage hook to capture this message sent to this process's window and handle it. I can define some custom message and handle for it. Is there any method to access this function in WinUI3? Thank you for your help.

**Comment by castorix** [comment]
With [SetWindowSubclass ](https://docs.microsoft.com/en-us/windows/win32/api/commctrl/nf-commctrl-setwindowsubclass)
